src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1577,6 +1606,97 @@ public override BoundNode VisitCall(BoundCall node)
             return null;
         }
 
+        private BoundExpression TryGetFirstArgument(MethodSymbol method, ImmutableArray<BoundExpression> arguments)
+        {
+            if (method.IsExtensionMethod)
+            {
```

Comment:
This does not seem correct. An extension method can still be invoked as an ordinary static method. Perhaps you are trying to check if it was invoked *as* an extension method? Or maybe this is really what you intend (check argument 0 of the method, or argument 1 if the first parameter has a `this` modifier)?